app/src/main/java/com/jalasoft/routesapp/ui/home/viewModel/HomeViewModel.kt

```diff
@@ -88,4 +95,71 @@ constructor(private val placeManager: PlaceRepository, private val gDirectionsRe
+
+    fun updateLocalLines(context: Context, history: SyncHistoryEntity) = viewModelScope.launch {
+        val resultList = routeRepository.searchForUpdatedLines(context, history.linesLastUpdated)
+        if (resultList.isNotEmpty()) {
+            for (item in resultList) {
+                routeLocalRepository.updateLocalLines(item)
+                updateLocalLineRoutes(item.idLine, history)
+            }
+            val updateHistory = SyncHistoryEntity(history.cityId, history.lineCategoryLastUpdated, Date().time, history.lineRoutesLastUpdated, history.TourPointCategoryLastUpdated, history.TourPointLastUpdated)
+            localDB.updateSyncHistory(updateHistory)
+        } else {
+            val currentCityId = PreferenceManager.getCurrentCityID(context)
+            val localList = routeLocalRepository.getAllLocalLinesByCityId(currentCityId)
+            for (item in localList) {
+                updateLocalLineRoutes(item.idLine, history)
+            }
+        }
+        updateTourPoints(context, history)
+    }
+
+    fun updateLocalLineRoutes(idLine: String, history: SyncHistoryEntity) = viewModelScope.launch {
+        val resultList = routeRepository.searchForUpdatedLineRoutes(idLine, history.lineRoutesLastUpdated)
+        if (resultList.isNotEmpty()) {
+            routeLocalRepository.updateLocalLineRoutes(resultList)
+            val updateHistory = SyncHistoryEntity(history.cityId, history.lineCategoryLastUpdated, history.linesLastUpdated, Date().time, history.TourPointCategoryLastUpdated, history.TourPointLastUpdated)
+            localDB.updateSyncHistory(updateHistory)
+        }
+    }
+
+    fun updateTourPointsCategory(history: SyncHistoryEntity) = viewModelScope.launch {
+        val resultList = tourPointsRepository.searchForUpdatedTourPointsCategory(history.TourPointCategoryLastUpdated)
+        if (resultList.isNotEmpty()) {
+            for (item in resultList) {
+                tourPointLocalRepository.updateLocalTourPointCategory(item)
+            }
+            val updateHistory = SyncHistoryEntity(history.cityId, history.lineCategoryLastUpdated, history.linesLastUpdated, history.lineRoutesLastUpdated, Date().time, history.TourPointLastUpdated)
+            localDB.updateSyncHistory(updateHistory)
+        }
+    }
+
+    fun updateTourPoints(context: Context, history: SyncHistoryEntity) = viewModelScope.launch {
+        val resultList = tourPointsRepository.searchForUpdatedTourPoints(context, history.TourPointLastUpdated)
+        if (resultList.isNotEmpty()) {
+            for (item in resultList) {
+                tourPointLocalRepository.updateLocalTourPoint(item)
+            }
+            val updateHistory = SyncHistoryEntity(history.cityId, history.lineCategoryLastUpdated, history.linesLastUpdated, history.lineRoutesLastUpdated, history.TourPointCategoryLastUpdated, Date().time)
+            localDB.updateSyncHistory(updateHistory)
+        }
+        updateTourPointsCategory(history)
+    }
```

Comment:
doesn't this should be added on their own Manager or repository? as they're not related to the viewmodel as is

Reply:
Now id done